### DISCUSSION TITLE
webのqiita editorのsyntax highlightによる不具合

### DISCUSSION THREAD

**Discussion by fygar256** [increments/qiita-discussions#897] [要望・アイデア・不具合]
webのqiitaのeditorで、記事に含まれるソースコードのsyntax highlightをonにすると、一部のデバイスでeditorの処理が重くなる不具合があるようです。大分以前からあり、対応してくれてないようなんですが、「一部のデバイスで」とあるので、自分のデバイスを調整すれば不具合が回避できるのではないかと思っています。私のマシンは、
OS: Arch Linux x86_64 
DE: Xfce 4.18 
GPU: AMD ATI Mobility Radeon HD 5650/5 
という構成です。
設定を変えたらいいかも知れませんが、どうすればいいか分かりません。
今は、不具合を起こす記事のsyntax hilightをoffにしています。

**Comment by fygar256** [comment]
一旦うまく行ってたのに、またうまく行きません。うまく行かないのは以下の３つの記事です。
https://qiita.com/fygar256/items/655bc44cc42f85c2f68f
https://qiita.com/fygar256/items/7b90aadfe5a80cb193f0
https://qiita.com/fygar256/items/35cd036ac38c4535a396

**Comment by fygar256** [comment]
自分のデバイスを調整すればうまくいくと思い、デバイスドライバをインストールしてみましたが、うまく動きませんでした。

**Comment by fygar256** [comment]
Qiitaをベータ版に切り替えたら、不具合の出る箇所が、Midnight Commanderの設定ファイルの箇所1箇所だけになったような気がします。

**Comment by tanaka-asami415** [comment]
@fygar256 
ご報告ありがとうございます。いただいた情報を元に調査を行いました。
一つのコードブロックに記述されるコードの量がかなり多い場合、重くなってしまう問題があるようです。

改善を検討していますが、修正されるまでは以下のいずれかの対応をしていただけると幸いです。

- plaintext 等を指定して syntax highlight を off にする
- 複数のコードブロックに分割して、1つのコードブロックに記述するコードの量を減らす
- [GitHub Gist](https://qiita.com/Qiita/items/612e2e149b9f9451c144#github-gist) などを使って記事にコードを埋め込む

ご不便をおかけいたしますが、何卒よろしくお願いいたします :bow:

**Comment by fygar256** [comment]
@tanaka-asami415 
どうもご対応ありがとうございました。

**Comment by fygar256** [comment]
一旦クローズしましたが、またreopenされてますね。本質的に問題が解決されてないからですね。
Qiita様も運営が大変でしょうね。
厄介なことはしたくないけど、しなければいけないのが仕事で、私はプロフェッショナルとしてエンジニアに従事したことがないので、
厳しさがわかってません。失礼の段は、何卒お許しをお願いします。

**Reply by tanaka-asami415** [reply]
クローズした場合、右側のEventsのところに表示が出ますので、こちらのDiscussionsはまだクローズいただいていないようです。
![image](https://github.com/user-attachments/assets/3539874d-b354-406a-9a85-ddf68de752b7)

もし、fygar256さんがクローズでもよければ、クローズをお願いいたします🙇

**Reply by fygar256** [reply]
どうもご返答ありがとうございます。
解決に至ってないようなので、openがいいです。

**Reply by tanaka-asami415** [reply]
承知しました。
大変恐れ入りますが、改善は検討しているものの、以下の理由から開発優先度は現時点ではかなり低くなっています。

- 今回の不具合は、一つのコードブロックに記述されるコードの量がかなり多い場合に現れていますが、同様の利用者があまりいない
- 記事の可読性などからコードを分ける方が多いので、量の多いコードが書けるようになった場合でも、利用される方は少なそう
- 代替策がある

もちろん、他の方からもリクエストをいただいた場合は、優先度を上げて検討したいと思います！

**Reply by fygar256** [reply]
わかりました。

**Reply by P-SiZK** [reply]
@fygar256
大変お待たせいたしました :bow:

以前ご報告いただいていた、コードブロックのコード量が多い場合にエディタが重くなる問題について、改善いたしましたのでご報告いたします。

入力中は一時的にシンタックスハイライトを適用せず、入力が止まったタイミングで適用するように変更しました。
これにより、動作が軽くなっているかと思います。

詳細はこちらのリリースノートをご参照ください。
https://qiita.com/release-notes/2026-07-14/improve-editor-performance
